### PR TITLE
Add test tooling and a basic test to go with it

### PR DESCRIPTION
**Pull request by geemanjs**
#7

### CODE REVIEW

**Comment by lennym** [comment]
I think `npm test` should run the _default_ tests - i.e. on built code, and a single run and there should be a different command for dev-time testing - i.e. on source code and with `--watch`.

I'd also advocate for keeping linting in the test script. 

`"test": "npm run test:lint && npm run test:unit"`

File: package.json [hunk 1]
```diff
@@ -4,9 +4,13 @@
   "description": "React components for govuk layouts and elements",
   "main": "index.js",
   "scripts": {
-    "test": "eslint .",
+    "lint": "eslint .",
+    "test": "jest --config ./scripts/jest/config.source.js --env=jsdom --watch",
+    "test:update": "npm run test -- --updateSnapshot",
```
Comment (lennym) [inline package.json]: What's this script for?

**Comment by lennym** [comment]
CI doesn't currently run *any* tests. It outputs the following message:

```
No tests found related to files changed since last commit.
```

I think CI should always run all tests.

**Comment by lennym** [comment]
Also it looks like CI is hanging and never terminating on account of the `--watch` flag.

File: package.json (hunk 1 above)
Comment (geemanjs) [reply package.json]: *tl;dr*
With snapshot testing the first time it executes it generates a snapshot of how the html should look. You can manually change the snapshot when you make a change that results in different html (that's perfect for small changes - i.e. most of the time). But sometimes its just easier to blat the original snapshot, generate a new one and validate the diff. That script generates the new one.

More detail: 
https://facebook.github.io/jest/docs/en/snapshot-testing.html

**Comment by geemanjs** [comment]
All to do with the `--watch` flag..

Should be fixed now (with a bit of luck)

**Comment by geemanjs** [comment]
I added tests for the other inputs.. noticed that none of them had the ability to have an 'onChange' handler so added this too.

In hindsight I probably should have split the onChange and test stuff into seperate PRs. But hey here we are!

**Comment by lennym** [comment]
This looks good. 

The reason there were no `on*` events bound to anything was just because my initial use case was only ever SSR and not ever actually running on the client. Thanks for adding.

File: package.json [hunk 2]
```diff
@@ -4,9 +4,15 @@
...
-    "test": "eslint .",
+    "lint": "eslint .",
+    "test": "npm run lint && npm run test:unit-built",
+    "test:unit-built": "jest --config ./scripts/jest/config.build.js --env=jsdom",
+    "test:unit-source": "jest --config ./scripts/jest/config.source.js --env=jsdom",
+    "test:watch": "npm run test:unit-source -- --watch",
+    "test:update": "npm run test -- --updateSnapshot",
     "build": "npm run build:js && npm run build:css",
-    "build:js": "babel ./src --out-dir ./components ",
+    "postbuild": "npm run test",
```
Comment (lennym) [inline package.json:14]: I'm not super thrilled with the idea of re-testing on _every_ build as it's going to be the kind of thing that's done _a lot_ during development, and so having to wait the extra 10 seconds on every build is likely to get annoying quickly.

I think I'd probably prefer to invert this order of control and run `npm run build` as a `pretest` script.

Comment (lennym) [reply package.json:14]: That said, I'm going to merge now, and if it gets horrific later it can change.

**Pull request merged**